storage2/src/storage.rs

```diff
@@ -12,13 +12,15 @@ use tracing::Span;
 use crate::snapshot::Snapshot;
 use crate::State;
 
+/// A handle for a storage instance, wrapping a RocksDB instance.
+#[derive(Clone)]
+pub struct Storage(Arc<Inner>);
```

Comment:
The same logic as https://github.com/penumbra-zone/penumbra/pull/1567/files#r1008648947 simplifies the implementation of committing new data to storage, and also gives an easy answer for how different threads access the `Storage` handle they use to create `State` forks.